src/commands/init.js

```diff
@@ -41,56 +99,17 @@ exports.builder = (yargs) => {
     })
 }
 
+exports.handlerTasks = handlerTasks;
+
 exports.handler = function ({ reporter, name, template }) {
   name = defaultAPMName(name)
   
   const basename = name.split('.')[0]
   const tasks = new TaskList([
-    {
-      title: 'Checking project existence',
-      task: async (ctx, task) => {
-        const projectPath = path.resolve(process.cwd(), basename)
-        const exists = await fs.pathExists(projectPath)
-        if (exists) {
-          throw new Error(`Couldn't initialize project. Project with name ${basename} already exists in ${projectPath}. Use different <name> or rename existing project folder.`)
-        }
-      }
-    },
-    {
-      title: 'Clone template',
-      task: async (ctx, task) => {
-        task.output = `Cloning ${template} into ${basename}...`
-
-        const repo = await clone(template, basename, { shallow: true })
-      }
-    },
-    {
-      title: 'Preparing template',
-      task: async (ctx, task) => {
-        // Set `appName` in arapp
-        const arappPath = path.resolve(
-          basename,
-          'arapp.json'
-        )
-        const arapp = await fs.readJson(arappPath)
-        arapp.appName = name
-
-        // Delete .git folder
-        const gitFolderPath = path.resolve(
-          basename,
-          '.git'
-        )
-
-        return Promise.all([
-          fs.writeJson(arappPath, arapp, { spaces: 2 }),
-          fs.remove(gitFolderPath)
-        ])
-      }
-    },
-    {
-      title: 'Install package dependencies',
-      task: async (ctx, task) => (await installDeps(basename, task)),
-    }
+    handlerTasks.checkProjectExistence({ basename }),
+    handlerTasks.cloneTemplate({ template, basename }),
+    handlerTasks.prepareTemplate({ name, basename }),
+    handlerTasks.installDependencies({ basename })
```

Comment:
I'd even take this one step further and do it similar to how Pando is doing it (as @sohkai mentioned), where there could be a `AragonCLI` library, that comes with all the tasks, which we can then use in CLI wrapper commands.

The library would be the core that'd be tested, while the CLI wrapper commands could be used for e2e tests.

Reply:
For reference: https://github.com/ryhope/pando/blob/master/packages/cli/src/commands/init/index.ts#L2

Reply:
I like that idea actually. Also I think the task title shouldn't be at library level (they are the CLI's UI), as it kind of is right now by being provided as a property to the task object (due to how TaskList works)

Reply:
Yes, I agree with that. We can roll that back. In fact, right now I'd vote for introducing a `lib` folder with all the functions that will be consumed by tasklist (and therefor CLI wrappers).

Once that is done, we can think about how to move away from Tasklist (if this is still something that should happen).